docs/installation.md

```diff
@@ -122,6 +122,42 @@ Create a Service with type `LoadBalancer` using the appropriate manifest for you
    nslookup <dns-name>
    ```
 
+## Upgrading NGINX Gateway Fabric
+
+### Upgrade NGINX Gateway Fabric from Manifests
+
+1. Upgrade the Gateway Resources
```

Comment:
if we upgrade GW API before NGF, that means NKG current version needs to support GW next version. Can we guarantee that?

Alternatively, If we upgrade NKG before GW, this means NKG next version must support the current GW API version.

I wonder what approach is easier to guarantee and also test?

Reply:
Can we assume that the GW API CRDs are backward compatible? They should be unless they rev the major version. I think in the case that there is a breaking change to the GW API CRDs, we will have to add special upgrade instructions to the release notes. But for the majority of cases, I think this method of upgrading is safe. 

I think upgrading NGF before GW is riskier, because NGF may depend on a new field in the latest version of the GW CRDs.

Reply:
> I think upgrading NGF before GW is riskier, because NGF may depend on a new field in the latest version of the GW CRDs.

I think there is a difference in the level of control too. For example, we can't control if a future version of the GW API can break us (there is a promise for backward compatibility from the GW API, but it is still a promise until tested after new version is released), but we can control what existing GW API versions a future NGF version can support (we can always test it before releasing ).